langchain/src/chat_models/llama_cpp.ts

```diff
@@ -298,11 +317,50 @@ export class ChatLlamaCpp extends SimpleChatModel<LlamaCppCallOptions> {
       .map((message) => {
         let messageText;
         if (message._getType() === "human") {
-          messageText = `[INST] ${message.content} [/INST]`;
+          switch (this.streamingModel) {
+            case "chatML":
+              messageText = `<|im_start|>user\n${message.content}<|im_end|>`;
+              break;
+            case "falcon":
+              messageText = `User: ${message.content}`;
+              break;
+            case "general":
+              messageText = `### Human\n${message.content}`;
+              break;
+            default:
+              messageText = `[INST] ${message.content} [/INST]`;
+              break;
```

Comment:
@nigel-daniels Why aren't you using a [`LlamaChatSession`](https://withcatai.github.io/node-llama-cpp/api/classes/LlamaChatSession) with the relevant `ChatPromptWrapper` for it?
For example, you have a `ChatPromptWrapper` for ChatML under [`ChatMLChatPromptWrapper`](https://withcatai.github.io/node-llama-cpp/api/classes/ChatMLChatPromptWrapper)
It's better to only have one centralized place for the handling of chat formats, especially as more will get supported in `node-llama-cpp` over time.

Furthermore, in the next major version of `node-llama-cpp` it'll figure out the right chat wrapper by default, so it won't require a manual config most of the time.

Reply:
This is because the call for streaming uses the `context.evaluate()` method:

```
const stream = await this.caller.call(async () =>
      this._context.evaluate(this._context.encode(prompt), promptOptions)
    );
```

This returned a generator function where I can use `yield` to pass back the tokens as they arrive.